rust/routee-compass/src/app/compass/config/compass_app_builder.rs

```diff
@@ -198,14 +199,15 @@ impl CompassAppBuilder {
 
     pub fn build_input_plugins(
         &self,
-        config: serde_json::Value,
+        mut config: serde_json::Value,
```

Comment:
again, we don't want to modify the upstream JSON configuration, we just want read access to it as a reference here. i also think this is a bit too far into the program to be modifying global configuration values, this will be easy to lose track of later.